// Copyright 2024 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto3";
package test.fuzzing.atspi_fuzzing;

// A fuzzer case for atspi_in_process_fuzzer.cc.
message FuzzCase {
  // Actions to take.
  repeated Action action = 1;
}

// Actions to take
message Action {
  // The path to the control with which we want to interact.
  // This doesn't give stable test cases - see comments in .cc file.
  repeated uint32 path_to_control = 1;

  // Wait until idle after the action.
  bool wait_afterwards = 2;

  oneof action_choice {
    // Take an "action" in an ATSPI sense, e.g. click a button
    TakeAction take_action = 3;

    // Edit some text.
    ReplaceText replace_text = 4;

    // Select items from a list.
    SetSelection set_selection = 5;
  }

  message TakeAction {
    uint32 action_id = 1;
  }

  message ReplaceText {
    string new_text = 1;
  }

  message SetSelection {
    repeated uint32 selected_child = 1;
  }
}
